File: php/OOP/Game.php
<?php

require_once __DIR__.'/Field.php';
require_once __DIR__.'/Config.php';

class Game
{
    public function __construct(Config $config, Field $field = null)
    {
        $this->field = $field;
        $this->config = $config;
    }

    public function setRandomField()
    {
        $this->field = new Field($this->config->get('width'), $this->config->get('height'));
        $this->field->populateRandomly();
    }

    public function play()
    {
        $i = 0;
        while (true) {
            echo "> $i".PHP_EOL.PHP_EOL;
            $this->field->output($this->config);

            $nextGenPoints = $this->field->getNextGenerationPoints();
            $this->field->setPoints($nextGenPoints);
            ++$i;
            usleep($this->config->get('timeout') * 1000);
            echo PHP_EOL.PHP_EOL.PHP_EOL;
        }
    }
}

File: java/src/Game.java
public class Game {
    public static String aliveStr;
    public static String deadStr;
    private int width;
    private int height;
    private int timeout;
    private Field field;

    public Game(int height, int width) {
        this.height = height;
        this.width = width;
        this.field = new Field(this.height, this.width);
    }
    
    public Game(Config config) {
        this.height = config.height;
        this.width = config.width;
        this.timeout = config.timeout;
        Game.aliveStr = config.aliveStr;
        Game.deadStr = config.deadStr;

        this.field = new Field(this.height, this.width);
    }

    public void play() {
        for (int i = 0; true; ++i) {
            System.out.printf("\n\nIteration %d\n\n", i);
            System.out.println(field.toString());

            this.field.update();

            this.waitTimeout();
        }
    }

    private void waitTimeout() {
        try {
            Thread.sleep(this.timeout);
        } catch (Exception exception) {

        }
    }
}